ndk/src/native_window.rs

```diff
@@ -39,11 +38,22 @@ impl Clone for NativeWindow {
     }
 }
 
-unsafe impl HasRawWindowHandle for NativeWindow {
-    fn raw_window_handle(&self) -> RawWindowHandle {
-        let mut handle = AndroidNdkWindowHandle::empty();
-        handle.a_native_window = self.ptr.as_ptr() as *mut c_void;
-        RawWindowHandle::AndroidNdk(handle)
+#[cfg(feature = "rwh_05")]
+unsafe impl rwh_05::HasRawWindowHandle for NativeWindow {
+    fn raw_window_handle(&self) -> rwh_05::RawWindowHandle {
+        let mut handle = rwh_05::AndroidNdkWindowHandle::empty();
+        handle.a_native_window = self.ptr.as_ptr().cast();
+        rwh_05::RawWindowHandle::AndroidNdk(handle)
+    }
+}
+
+#[cfg(feature = "rwh_06")]
+impl rwh_06::HasWindowHandle for NativeWindow {
+    fn window_handle(&self) -> Result<rwh_06::WindowHandle<'_>, rwh_06::HandleError> {
+        let handle = rwh_06::AndroidNdkWindowHandle::new(self.ptr.cast());
+        let handle = rwh_06::RawWindowHandle::AndroidNdk(handle);
+        // SAFETY: All fields of the "raw"  `AndroidNdkWindowHandle` struct are filled out
```

Comment:
Nit: you should add a safety comment about how the handle is guaranteed to be valid.

Reply:
I expected to have to deal with this, but the documentation only seems to discuss nullability of the handles since such "validity claims" should be "irrelevant" given the lifetime?

https://docs.rs/raw-window-handle/latest/raw_window_handle/struct.WindowHandle.html#safety

Unless I'm misunderstanding your docs, which somehow seem to jump from nullability to vailidity of handles within the same sentence?

Reply:
Yeah that's my bad. Just add something to the tune of "this pointer will always be valid."

Reply:
Sure, done.